src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2757,6 +2757,86 @@ static QVariant fcnGeometryCollectionAsArray( const QVariantList &values, const
   return array;
 }
 
+static QVariant fcnGeometryDissolve( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom;
+  QVector<QgsGeometry> geometries;
+  for ( int i = 0; i < values.count(); i++ )
+  {
+    geom = QgsExpressionUtils::getGeometry( values.at( i ), parent );
+    if ( geom.isNull() )
+      continue;
+    geometries.append( geom.asGeometryCollection() );
+  }
+  QgsGeometry dissolved = QgsGeometry().unaryUnion( geometries );
+  return QVariant::fromValue( dissolved );
+}
+
+static QVariant fcnSplitGeometry( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry initGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  QgsGeometry inGeom = initGeom;
+  if ( inGeom.isNull() || inGeom.type() == QgsWkbTypes::PointGeometry )
+    return QVariant::fromValue( initGeom );
+  QgsGeometry splitGeom = QgsExpressionUtils::getGeometry( values.at( 1 ), parent );
+  if ( splitGeom.isNull() || splitGeom.type() == QgsWkbTypes::PointGeometry )
```

Comment:
It's possible to split eg a line by a point -- is there a particular reason you've explicitly prevented this?

Reply:
I was assuming that there could be some precision issues and thought it was not possible, so I opted on returnin sooner, I can remove the check.